plugins/filter_aws/aws.c

```diff
@@ -434,7 +656,36 @@ static struct flb_config_map config_map[] = {
      " will be used: 'v1' or 'v2'. 'v2' may not work"
      " if you run Fluent Bit in a container."
     },
-
+    {
+     FLB_CONFIG_MAP_BOOL, "ec2_instance_type", "false",
+     0, FLB_FALSE, 0,
+     "Enable EC2 instance type"
+    },
+    {
+     FLB_CONFIG_MAP_BOOL, "private_ip", "false",
+     0, FLB_FALSE, 0,
+     "Enable EC2 instance private ip"
+    },
```

Comment:
Check out a newer config map, for example here in S3: https://github.com/fluent/fluent-bit/blob/master/plugins/out_s3/s3.c#L1339

Note the `offsetof`. You can use this to set the configuration properties on the context, which means you can get ride of the special function you created for that. 

In this case, you want something like:
```
{
     FLB_CONFIG_MAP_BOOL, "private_ip", "false",
     0, FLB_TRUE, offsetof(struct flb_filter_aws, private_ip_include),
     "Enable EC2 instance private ip"
    },
```

Reply:
Thanks for sharing the example! Changed to using `offsetof`.